manifests/http_port.pp

```diff
@@ -1,24 +1,56 @@
 define squid::http_port (
-  Variant[Pattern[/\d+/], Integer]
-          $port    = $title,
-  Boolean $ssl     = false,
-  String  $options = '',
-  String  $order   = '05',
+  Optional[Stdlib::Port] $port    = undef,
+  Optional[Stdlib::Host] $host    = undef,
+  Boolean                $ssl     = false,
+  String                 $options = '',
+  String                 $order   = '05',
 ) {
+  $_title = String($title)
```

Comment:
I'm not sure of this, but aren't titles always strings?

Reply:
I was pretty sure I had been bitten by an Integer title when I was writing the first version of this PR, 6 months ago. Hence I took the extra care about this otherwise non-obvious issue.

However, a little research shows the title must be a string since a [2014 commit](https://github.com/puppetlabs/puppet/commit/d25e462ebe76f3f7a4391717ff084a91e3005b7d#diff-9c2c9ff9f4b472b68b4fc5155ce9d98bR696) associated with [Puppet 3.7.0](https://tickets.puppetlabs.com/browse/PUP-2898). And I definitively haven't been near a Puppet 3 for years.

I can look at it more closely next week if you want. That `$_title` var is ugly.

Reply:
I thought about it some more and I cannot see a scenario where String-casting `$title` is useful, so I removed it.

Reply:
Ah, this is why I added the String-cast! `create_resources()` bypasses the check done in the commit I linked above. So yes, you can have Integer titles (although it's probably a bug :).

I'll revert to the 486e869 version.

Reply:
See, I knew it was supposed to be that way but there was probably a reason you put it there :)